controllers/sessionController.js

```diff
@@ -0,0 +1,83 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+const csrf = require("host-csrf");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+// const registerDo = async (req, res, next) => {
+//   if (req.body.password != req.body.password1) {
+//     req.flash("error", "The passwords entered do not match.");
+//     res.render("register", { errors: req.flash.errors });
+//   }
+//   try {
+//     await User.create(req.body);
+//   } catch (e) {
+//     console.error(e);
+//     if (e.constructor.name === "ValidationError") {
+//       parseVErr(e, req);
+//     } else if (e.name === "MongoServerError" && e.code === 11000) {
+//       req.flash("error", "That email address is already registered.");
+//     } else {
+//       return next(e);
+//     }
+//     return res.render("register");
+//   }
+//   res.redirect("/");
+// };
+
+const registerDo = async (req, res, next) => {
+  let validation_errors = false;
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    validation_errors = true;
+  }
```

Comment:
You'll need to add an else here, and the following code should be included inside the else. Because as it is, even when the password doesn't match, the application will continue registering the user.